core/include/gnuradio-4.0/Block.hpp

```diff
@@ -1513,10 +1513,21 @@ class Block : public lifecycle::StateMachine<Derived>, public std::tuple<Argumen
             emitErrorMessageIfAny("workInternal(): EOS tag arrived -> REQUESTED_STOP", this->changeStateTo(lifecycle::State::REQUESTED_STOP));
             publishEoS();
             this->setAndNotifyState(lifecycle::State::STOPPED);
-            return {requested_work, 0UZ, DONE};
+            return {requestedWork, 0UZ, DONE};
         }
+
         if (asyncEoS || (resampledIn == 0 && resampledOut == 0 && !hasAsyncIn && !hasAsyncOut)) {
-            return {requested_work, 0UZ, resampledStatus};
+            return {requestedWork, 0UZ, resampledStatus};
+        }
+
+        // Apply soft cut on requestedWork
+        if (availableToProcess > requestedWork) {
+            // include requestedWork in computeResampling calculation
+            const auto [resampledIn2, resampledOut2, resampledStatus2] = computeResampling(std::min(minSyncIn, nextEosTag), std::min(availableToProcess, requestedWork), minSyncOut, availableToPublish);
```

Comment:
The clamping looks OK but doesn't handle the corner case where requestedWork is < minSyncIn. The requestedWork clamping should not suppress the block to process data if it would have been able to if the requested work would have been max.

Reply:
Otherwise OK.